src/common/procedure/src/local/runner.rs

```diff
@@ -147,6 +154,11 @@ impl Runner {
         // Release locks and notify parent procedure.
         guard.finish();
 
+        // If `ProcedureManager` is stopped, it stops the current task immediately without deleting the procedure.
+        if !self.running() {
```

Comment:
When we reach here, the procedure is already finished. I think we can still clean up resources in memory to avoid leaking them. We can move this check after L166 `self.manager_ctx.on_procedures_finish(&procedure_ids);`